tests/vm_watch_test.go

```diff
@@ -21,338 +24,380 @@ const (
 	relevantk8sVer = "1.16.2"
 
 	// Define a timeout for read opeartions in order to prevent test hanging
-	readTimeout     = 1 * time.Minute
-	processWaitTime = 2 * time.Minute
+	readTimeout       = 1 * time.Minute
+	vmCreationTimeout = 30 * time.Second
 
 	bufferSize = 1024
 )
 
-var _ = Describe("[rfe_id:3423][crit:high][vendor:cnv-qe@redhat.com][level:component]VmWatch", func() {
-	tests.FlagParse()
-
-	virtCli, err := kubecli.GetKubevirtClient()
-	tests.PanicOnError(err)
-
-	type vmStatus struct {
-		name,
-		age,
-		running,
-		volume string
+type vmStatus struct {
+	name,
+	age,
+	running,
+	volume string
+}
+
+type vmiStatus struct {
+	name,
+	age,
+	phase,
+	ip,
+	node string
+}
+
+func newVMStatus(fields []string) *vmStatus {
+	flen := len(fields)
+	stat := &vmStatus{}
+
+	switch {
+	case flen > 3:
+		stat.volume = fields[3]
+		fallthrough
+	case flen > 2:
+		stat.running = fields[2]
+		fallthrough
+	case flen > 1:
+		stat.age = fields[1]
+		fallthrough
+	case flen > 0:
+		stat.name = fields[0]
 	}
 
-	type vmiStatus struct {
-		name,
-		age,
-		phase,
-		ip,
-		node string
+	return stat
+}
+
+func newVMIStatus(fields []string) *vmiStatus {
+	flen := len(fields)
+	stat := &vmiStatus{}
+
+	switch {
```

Comment:
@omeryahud I really don't get this test. If I start until a VMI is waiting, I should get a pretty well defined response. Why is there all this extra parsing going on. Do you know?

Reply:
The point of this test is to make sure that the end user receives all VM or VMI updates when using kubectl get vm/vmi -w.
Parsing the response to vmiStatus and vmStatus objects is mainly for readability instead of asserting on slices

Reply:
Hm, regarding to readability. I think I would have preferred if the test just compares it to a string which uses `fmt.Spprintf()` to insert the expected node and the expected IP. I think that is much simpler and makes it much easier to follow.

Reply:
String comparison won't work since the number of white space characters between columns is not constant, so I will have to use `strings.Fields()` (which I already do right now) and mess around with slices, that's why I added the helper structs to give it some more context.

Reply:
how about something like this:

```go
expected := []string{"vmi-nocloud"   2m   Failed   10.244.1.25   node02}
fields := strings.Fields(line)
Expect(fields).Should(ConsistOf("vmi-nocloud",MatchRegex("[0-9]+[ms]"),"Failed", ip, node))
```